### PR TITLE
Add some core modules and new features 

### PR DESCRIPTION
**Pull request by ai-chen2050**
This pull request mainly contains the following features:

1. feat: add the common tee enclave utilities, and vlc in tee
2. feat: add the vrf implementation crate
3. feat: add the crypto crate
4. refac: rename some unused variant names

## Release Notes

- **New Features**
  - Introduced multiple new crates: `vrf`, `crypto`, `enclaves`, and `types` to enhance modularity and functionality.
  - Added a `LICENSE` file adopting the MIT License for clear usage terms.
  - Enhanced documentation with new sections in `crates/README.md` for `enclaves`, `crypto`, and `vrf` modules.
  - Implemented a new `NitroSecureModule` for secure communication with AWS Nitro Enclaves.
  - Added a `NitroEnclavesClock` for clock synchronization within secure environments.
  - Introduced the `tee_vlc` module for verifiable logic clocks in a TEE context.

- **Bug Fixes**
  - Renamed demo entries for consistency and clarity.

- **Documentation**
  - Updated `demos/README.md` to include the new `tee_vlc` module and its features.
  - Added comprehensive README for `tee_vlc` detailing setup and execution in a TEE context.

- **Style**
  - Improved naming conventions across various modules for better clarity.

- **Tests**
  - Introduced extensive unit tests for the VRF implementation to ensure reliability and security.

### CODE REVIEW

File: crates/enclaves/src/nitro_secure.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            nonce: None,
+            public_key: None,
+        };
+        let Attestation {
+            user_data: Some(buf),
+            ..
+        } = &mut request
+        else {
+            unreachable!()
+        };
```
[the remark this answers is not in the document]
Comment (ai-chen2050) [reply crates/enclaves/src/nitro_secure.rs]: user_data is a private field

File: crates/enclaves/src/nitro_secure.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            loop {
+                let result = tokio::select! {
+                    result = &mut read_session, if !read_session.is_finished() => result,
+                    result = &mut write_session, if !write_session.is_finished() => result,
+                    else => break,
+                };
+                if let Err(err) = result.map_err(Into::into).and_then(std::convert::identity) {
+                    warn!("{err}")
+                }
+            }
```
[the remark this answers is not in the document]
Comment (ai-chen2050) [reply crates/enclaves/src/nitro_secure.rs:147]: this loop keeps one connect, and still works when meets some error in only connect.

File: demos/tee_vlc/src/bin/call_vlc_client.rs [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            }),
+        )
+    };
+
+    let (portal_session, session) = run_nitro_client;
+    'select: {
+        tokio::select! {
+            result = session => break 'select result??,
+            result = portal_session => result??,
+        }
```
[the remark this answers is not in the document]
Comment (ai-chen2050) [reply demos/tee_vlc/src/bin/call_vlc_client.rs:102]: Double wrapper error, so ?? is ok in there.

File: crates/vrf/src/traits.rs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+// pub trait SeedableCryptoRng = ::rand::SeedableRng + ::rand::RngCore + ::rand::CryptoRng;
+
+/// A type family for schemes which know how to generate key material from
+/// a cryptographically-secure [`CryptoRng`][::rand::CryptoRng].
+pub trait Uniform {
+    /// Generate key material from an RNG for testing purposes.
+    fn generate_for_testing<R>(rng: &mut R) -> Self
+    where
+        R: ::rand::SeedableRng + ::rand::RngCore + ::rand::CryptoRng;
+}
```
[the remark this answers is not in the document]
Comment (ai-chen2050) [reply crates/vrf/src/traits.rs:215]: Its just be used to testing

**Comment by bufrr** [comment]
@coderabbitai full review

**Review by fshif — APPROVED** [review]

**Pull request merged**